src/sleeplocks.erl

```diff
@@ -143,8 +143,8 @@ handle_call(release, {From, _Ref}, #lock{current = Current} = Lock) ->
     NewLock = case maps:is_key(From, Current) of
         false -> Lock;
         true  ->
-            NewCurrent = maps:remove(From, Current),
-            next_caller(Lock#lock{current = NewCurrent})
+            erlang:demonitor(map:get(From, Current)),
+            next_caller(Lock#lock{current = maps:remove(From, Current)})
```

Comment:
Can you please change this back to:

```
NewCurrent = maps:remove(From, Current),
erlang:demonitor(NewCurrent),
next_caller(Lock#lock{current = NewCurrent})
```

There's no point looking up inside the map twice.

Reply:
That actually would not work, because `maps:remove/2` and `maps:get/2` do not return the same thing. The call in my version of the code returns the Reference to the monitor, which is needed to remove. When creating the lock I store that as value in the map (instead of the `ok` it used to be).

But, I see that I made a typo, `map:get/2` vs `maps:get/2`. So much for manual testing. Will look into that.

Reply:
Fixed the typo, left the rest in, curious to know what you think with this context.

Reply:
You're right! That's what I get for writing on my phone.

It should actually use `maps:take/2` at this point, instead of `maps:remove/2`. Would you like to update to use that?

Reply:
`maps:take/2` is indeed better, updated it. Also worked on a little (uncommitted) test script that runs successful, for greater confidence :)

`erlc src/sleeplocks.erl && escript test.escript`

```erlang

main([]) ->
  Lock = {global, lock},
  sleeplocks:new(1, [{name, Lock}]),

  %% local locks
  ok = sleeplocks:attempt(Lock),
  {error, unavailable} = sleeplocks:attempt(Lock),
  ok = sleeplocks:release(Lock),

  %% remote locks once
  ok = sleeplocks:attempt(Lock),
  spawn_link(fun() ->
    {error, unavailable} = sleeplocks:attempt(Lock)
  end),
  ok = sleeplocks:release(Lock),

  %% remote locks twice
  Pid = spawn(fun() ->
    ok = sleeplocks:attempt(Lock),
    receive bye -> exit(boom) end
  end),
  true = erlang:is_process_alive(Pid),
  {error, unavailable} = sleeplocks:attempt(Lock),
  Pid ! bye,
  timer:sleep(100),
  false = erlang:is_process_alive(Pid),
  ok = sleeplocks:attempt(Lock),

  io:format("success\n").
```